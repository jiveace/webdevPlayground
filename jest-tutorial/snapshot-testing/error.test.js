import getErrorMessage from "./error";

describe("getErrorMessage", () => {
  it("returns message for a valid code", () => {
    expect(getErrorMessage(1)).toMatchSnapshot();
    expect(getErrorMessage(2)).toMatchSnapshot();
    expect(getErrorMessage(3)).toMatchSnapshot();
  });

  it("throws an error otherwise", () => {
    expect(() => getErrorMessage(4)).toThrowErrorMatchingSnapshot();
  });
});
