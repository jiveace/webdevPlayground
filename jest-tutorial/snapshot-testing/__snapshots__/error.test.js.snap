// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getErrorMessage returns message for a valid code 1`] = `"Oh Sphinx of black quartz, judge my vow"`;

exports[`getErrorMessage returns message for a valid code 2`] = `"Rabbits don't eat carrots"`;

exports[`getErrorMessage returns message for a valid code 3`] = `"Cats don't eat mouses"`;

exports[`getErrorMessage throws an error otherwise 1`] = `"No error messages for that code"`;
